src/policy/rbf.h

```diff
@@ -31,4 +36,51 @@ enum class RBFTransactionState {
 RBFTransactionState IsRBFOptIn(const CTransaction& tx, const CTxMemPool& pool) EXCLUSIVE_LOCKS_REQUIRED(pool.cs);
 RBFTransactionState IsRBFOptInEmptyMempool(const CTransaction& tx);
 
+/** Get all descendants of iters_conflicting. Also enforce BIP125 Rule 5 and quit as early as
+ * possible. There cannot be more than MAX_BIP125_REPLACEMENT_CANDIDATES potential entries.
+ * @param[in]   iters_conflicting   The set of iterators to mempool entries.
+ * @param[out]  state               Used to return errors, if any.
+ * @param[out]  all_conflicts       Populated with all the mempool entries that would be replaced,
+ *                                  which includes descendants of iters_conflicting. Not cleared at
+ *                                  the start; any existing mempool entries will remain in the set.
+ * @returns false if Rule 5 is broken.
+ */
+bool GetEntriesForRBF(const CTransaction& tx, CTxMemPool& pool,
+                      const CTxMemPool::setEntries& iters_conflicting, TxValidationState& state,
+                      CTxMemPool::setEntries& all_conflicts) EXCLUSIVE_LOCKS_REQUIRED(pool.cs);
+
+/** Enforce BIP125 Rule 2: a replacement transaction must not add any new unconfirmed inputs. */
+bool HasNoNewUnconfirmed(const CTransaction& tx, const CTxMemPool& pool,
+                         const CTxMemPool::setEntries& iters_conflicting,
+                         TxValidationState& state) EXCLUSIVE_LOCKS_REQUIRED(pool.cs);
+
+/** Check the intersection between two sets of mempool entries to make sure they are disjoint.
+ * @param[in]   ancestors           Set of mempool entries corresponding to ancestors of the
+ *                                  replacement transactions.
+ * @param[in]   direct_conflicts    Set of mempool entries corresponding to the mempool conflicts
+ *                                  (candidates to be replaced).
+ * @param[in]   hash                Transaction ID, included in the error message if violation occurs.
+ * returns true if the two sets are disjoint (i.e. intersection is empty), false if otherwise.
+ */
+bool SpendsAndConflictsDisjoint(const CTxMemPool::setEntries& ancestors,
```

Comment:
I didn't end up renaming `hash` to `txid` because I want to be able to pass in either package id or txid in the future :thinking: let me know if it's too weird though, and I can change it

Reply:
I'd personally prefer fewer surprise tools that help us later, since it's easy enough to update comments/rename parameters when you actually need that functionality. No big deal either way though!

Reply:
renamed to txid